packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3599,9 +3600,42 @@ export class TransactionController extends BaseController<
       );
     }
 
+    if (this.#checkIfSimulationRetriggerNeeded(transactionMeta)) {
+      this.#updateSimulationData(transactionMeta).catch((error) => {
+        log('Error updating simulation data', error);
+        throw error;
+      });
+    }
+
     return transactionMeta;
   }
 
+  #checkIfSimulationRetriggerNeeded(transactionMeta: TransactionMeta) {
```

Comment:
Won't this keep firing unless we also pass the `previousTransactionMeta` and also check if any of the relevant inputs have changed?

Reply:
Good spot Matthew, this is now fixed, comparing before and after